x/leverage/keeper/keeper_test.go

```diff
@@ -128,6 +147,279 @@ func (suite *IntegrationTestSuite) TestWithdrawAsset_Valid() {
+	suite.Require().NoError(err)
+
+	// verify lender's new loan amount (120 umee)
+	loanBalance := app.LeverageKeeper.GetLoan(ctx, lenderAddr, umeeapp.BondDenom)
+	suite.Require().Equal(loanBalance, tCoin("umee", 120))
+
+	// verify lender's new umee balance (10 - 1k from initial + 200 from loan - 80 repaid = 9120)
+	tokenBalance := app.BankKeeper.GetBalance(ctx, lenderAddr, umeeapp.BondDenom)
+	suite.Require().Equal(tCoin("umee", 9120), tokenBalance)
+
+	// verify lender's uToken balance remains at 1k from initial conditions
+	uTokenBalance := app.BankKeeper.GetBalance(ctx, lenderAddr, "u/"+umeeapp.BondDenom)
+	suite.Require().Equal(tCoin("u/umee", 1000), uTokenBalance)
+
+	// lender repays 120 umee (loan repaid in full)
+	err = app.LeverageKeeper.RepayAsset(ctx, lenderAddr, tCoin("umee", 120))
+	suite.Require().NoError(err)
+
+	// verify lender's new loan amount in the correct denom (zero)
+	loanBalance = app.LeverageKeeper.GetLoan(ctx, lenderAddr, umeeapp.BondDenom)
+	suite.Require().Equal(loanBalance, tCoin("umee", 0))
+
+	// verify lender's new umee balance (10 - 1k from initial + 200 from loan - 200 repaid = 9000)
+	tokenBalance = app.BankKeeper.GetBalance(ctx, lenderAddr, umeeapp.BondDenom)
+	suite.Require().Equal(tCoin("umee", 9000), tokenBalance)
+
+	// verify lender's uToken balance remains at 1k from initial conditions
+	uTokenBalance = app.BankKeeper.GetBalance(ctx, lenderAddr, "u/"+umeeapp.BondDenom)
+	suite.Require().Equal(tCoin("u/umee", 1000), uTokenBalance)
+}
+
+func (suite *IntegrationTestSuite) TestRepayAsset_Overpay() {
+	lenderAddr, _ := suite.initBorrowScenario()
+	app, ctx := suite.app, suite.ctx
+
+	// The "lender" user from the init scenario is being used because it
+	// already has 1k u/umee for collateral
+
+	// lender borrows 200 umee
+	err := app.LeverageKeeper.BorrowAsset(ctx, lenderAddr, tCoin("umee", 200))
+	suite.Require().NoError(err)
+
+	// lender repays 300 umee - should automatically reduce to 200 (the loan amount) and succeed
+	err = app.LeverageKeeper.RepayAsset(ctx, lenderAddr, tCoin("umee", 300))
+	suite.Require().NoError(err)
+
+	// verify lender's new loan amount is zero
+	loanBalance := app.LeverageKeeper.GetLoan(ctx, lenderAddr, umeeapp.BondDenom)
+	suite.Require().Equal(loanBalance, tCoin("umee", 0))
+
+	// verify lender's new umee balance (10 - 1k from initial + 200 from loan - 200 repaid = 9000)
+	tokenBalance := app.BankKeeper.GetBalance(ctx, lenderAddr, umeeapp.BondDenom)
+	suite.Require().Equal(tokenBalance, tCoin("umee", 9000))
+
+	// verify lender's uToken balance remains at 1k from initial conditions
+	uTokenBalance := app.BankKeeper.GetBalance(ctx, lenderAddr, "u/"+umeeapp.BondDenom)
+	suite.Require().Equal(uTokenBalance, tCoin("u/umee", 1000))
+
+	// lender repays 50 umee - this time it fails because the loan no longer exists
+	err = app.LeverageKeeper.RepayAsset(ctx, lenderAddr, tCoin("umee", 50))
+	suite.Require().Error(err)
+}
```

Comment:
Would it be okay to keep some redundant comments inside the _test files, when they're inside functions where the majority of blocks have meaningful comments?

For example in the highlighted function, I know `// lender borrows 200 umee` has no need to be a comment, but its presence makes the test function itself very stylistically consistent (each meaningful action is a whitespace-separated code block with a leading comment, followed by actions then additional checks, if any).